jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriter.java

```diff
@@ -134,13 +142,34 @@ public void doWrite(Server server, Query query, ImmutableList<Result> results) t
 		// Creates only if it doesn't already exist
 		influxDB.createDatabase(database);
 
+		// use alias if provided, otherwise hostname
+		String source = getSource(server);
+
 		BatchPoints batchPoints = BatchPoints.database(database).retentionPolicy(retentionPolicy)
-				.tag(TAG_HOSTNAME, server.getHost()).consistency(writeConsistency).build();
+				.tag(TAG_HOSTNAME, source).consistency(writeConsistency).build();
+
 		for (Result result : results) {
-			Map<String, String> resultTagsToApply = buildResultTagMap(result);
-			Point point = Point.measurement(result.getKeyAlias()).time(result.getEpoch(), MILLISECONDS)
-					.tag(resultTagsToApply).fields(result.getValues()).build();
-			batchPoints.point(point);
+			// we'll create a copy of result values here
+			Map<String, Object> fixedValues = new HashMap<String,Object>();
+			// _jmx_port as a field so we can filter on it in influx
+			fixedValues.put("_jmx_port", Integer.parseInt(server.getPort()));
+			// clean up an NaN values in values
+			Map<String, Object> resultValues = result.getValues();
+			if (resultValues != null) {
```

Comment:
`result.getValues()` is tagged ` @Nonnull`, so the null check should be removed (I know, it was there already, but code should be cleaner every time we touch it...)